File: Complexos.java
import java.util.Scanner;

public class Complexos {
    private int real;
    private int imaginario;

    public Complexos(int real, int imaginario) {
        this.real = real;
        this.imaginario = imaginario;
    }

    public Complexos soma(Complexos outro) {
        return new Complexos(this.real + outro.real, this.imaginario + outro.imaginario);
    }

    public Complexos subtrai(Complexos outro) {
        return new Complexos(this.real - outro.real, this.imaginario - outro.imaginario);
    }

    public Complexos multiplica(Complexos outro) {
        int r = this.real * outro.real - this.imaginario * outro.imaginario;
        int i = this.real * outro.imaginario + this.imaginario * outro.real;
        return new Complexos(r, i);
    }

    public Complexos divide(Complexos outro) {
        int denominador = outro.real * outro.real + outro.imaginario * outro.imaginario;
        int r = (this.real * outro.real + this.imaginario * outro.imaginario) / denominador;
        int i = (this.imaginario * outro.real - this.real * outro.imaginario) / denominador;
        return new Complexos(r, i);
    }

    @Override
    public String toString() {
        return this.real + (this.imaginario >= 0 ? " +" : " ") + this.imaginario + "i";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNext()) {
            int a = scanner.nextInt();
            String biStr = scanner.next();
            int b = Integer.parseInt(biStr.replace("i", ""));
            String operador = scanner.next();
            int c = scanner.nextInt();
            String diStr = scanner.next();
            int d = Integer.parseInt(diStr.replace("i", ""));

            Complexos m1 = new Complexos(a, b);
            Complexos m2 = new Complexos(c, d);
            Complexos resultado = null;

            switch (operador) {
                case "+":
                    resultado = m1.soma(m2);
                    break;
                case "-":
                    resultado = m1.subtrai(m2);
                    break;
                case "*":
                    resultado = m1.multiplica(m2);
                    break;
                case "/":
                    resultado = m1.divide(m2);
                    break;
           }

            System.out.println(resultado);
        }
    }
}

File: Complexos.cpp
#include <iostream>
using namespace std;

class Complexo {
private:
    int real;
    int imaginario;

public:
    Complexo(int r=0; int i=0){
        real = r;
        imaginario = i;
    }

    Complexo operator+(const Complexo& c) const {
        return Complexo(real + c.real, imaginario + c.imaginario);
    }

    Complexo operator-(const Complexo& c) const {
        return Complexo(real - c.real, imaginario - c.imaginario);
    }

    Complexo operator*(const Complexo& c) const {
        int r = real * c.real - imaginario * c.imaginario;
        int i = real * c.imaginario + imaginario * c.real;
        return Complexo(r, i);
    }

    Complexo operator/(const Complexo& c) const {
        int denominador = c.real * c.real + c.imaginario * c.imaginario;
        int r = (real * c.real + imaginario * c.imaginario) / denominador;
        int i = (imaginario * c.real - real * c.imaginario) / denominador;
        return Complexo(r, i);
    }

    void exibir() const {
        cout << real << "+" << imaginario << "i" << endl;
    }
};

int main(){
    int a, b, c, d;
    char op;

    while (cin >> a >> b >> op >> c >> d) {
        Complexo z1(a, b);
        Complexo z2(c, d);
        Complexo resultado;

        switch (op) {
            case '+':
                resultado = z1 + z2;
                break;
            case '-':
                resultado = z1 - z2;
                break;
            case '*':
                resultado = z1 * z2;
                break;
            case '/':
                resultado = z1 / z2;
                break;
        }

        resultado.exibir();
    }
    return 0;
}